MD5 out... in with SHA256

kensipe opened the pull request:
Signed-off-by: Ken Sipe <kensipe@gmail.com>


**Description of the change:**
Move away from MD5 which the linter is angry with...

```
internal/olm/operator/internal/configmap.go:19:2: G501: Blacklisted import `crypto/md5`: weak cryptographic primitive (gosec)
	"crypto/md5"
	^
internal/olm/operator/internal/configmap.go:125:7: G401: Use of weak cryptographic primitive (gosec)
	h := md5.New()
	     ^
```

**Motivation for the change:**
Making the linter happy

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

camilamacedo86 commented on internal/olm/operator/internal/configmap.go:
```diff
@@ -122,7 +122,7 @@ func makeObjectFileName(b []byte, names ...string) string {
 
 // hashContents creates a base32-encoded md5 digest of b's bytes.
```
we would need to change the comment as well.

kensipe commented:
@camilamacedo86 I've rebased and added gosec for this fix only... excluding all gosec other rules for now.  there was an open slack discussion that this was a desired solution... link:  https://kubernetes.slack.com/archives/C017UU45SHL/p1596812929074800

good catch on comment!

joelanford commented:
Thinking through compatibility...

If someone ran `operator-sdk run packagemanifests` before this change, didn't cleanup, then upgraded their SDK version and ran `operator-sdk run packagemanifests` again after this change, the only not-necessary-but-also-non-breaking thing we would do is regenerate the configmaps because we would think that they were stale due to a different hash?

Any other potential side-effects that anyone can think of that would make this a breaking change? We have to be extra careful now :)

camilamacedo86 commented on .golangci.yml:
```diff
@@ -17,4 +17,9 @@ linters:
+  exclude-rules:
+    - linters:
+      - gosec
+      text: "G110|G601|G404|G204|G306"
```
I am not in favor to add gosec because I have been testing it and in POV it has not raised nice things and raise a false positive as well.

camilamacedo86 commented on internal/olm/operator/internal/configmap.go:
```diff
@@ -120,9 +120,9 @@ func makeObjectFileName(b []byte, names ...string) string {
+// hashContents creates a sha256 digest of b's bytes.
 func hashContents(b []byte) string {
-	h := md5.New()
+	h := sha256.New()
```
For example here, why the md5 would be a risk in this case? Why we should change to `sha256`? Shows that we have no reason to do this change at all. the only reason here is that the CI pass in the gosec check,

kensipe replied:
@camilamacedo86 this is not true... This change was not put in place to satisfy the linter... md5 has been reported as broken and unsuitable since 2010.   sha256 is a good recommended solution. 

https://www.md5online.org/blog/why-md5-is-not-safe/

camilamacedo86 commented:
Hi @kensipe, 

IMO we should not move forward with this one. Also, see the @joelanford comments https://github.com/operator-framework/operator-sdk/pull/3683#issuecomment-673604662. 

In this way, I hope that you are ok with we close this one. However, please feel free to continue contributing to the project and re-open it in case you do not agree. Your PR's are very welcome.

kensipe commented:
@camilamacedo86 @joelanford comment was not a down vote... it was an observation and a request for more feedback / concerns.  I specifically put this PR together because it was discussed on a slack thread between @estroz @joelanford and myself and this was the stated direction.    Now I'm confused... is this the desired direction?   I would definitely switch from md5.   I'll rebase

camilamacedo86 commented:
Hi @kensipe, 

I do not see an advantage to add this check. However, feel free to re-open if you wish and if the others are ok with then, all fine for me as well.

camilamacedo86 commented:
/reopen
